hw3/02-charts.js

```diff
@@ -29,24 +31,123 @@ const borderColors = [
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
 
-const renderChart = () => {
+
+// async function to fetch the data from a given url
+const fetchData = async function fetchDataFromApi(url) {
+  try {
+      const res = await fetch(url);
+      const data = await res.json();
+      return data;
+  }
+  catch {
+      console.error('Error', error);
+  }
+};
+
+// function to count family names
+const getFam = function getFamilyNames(data) {
+
+  let familyNames = [];
+
+  data.forEach(element => {
+    if(!familyNames.some(name => name.name === element.lastName)) {
```

Comment:
The API has two values for characters, 'lastName' which gives the character's last name and 'family' which is their family's name, I personally used family but I'm not sure off the top of my head if there are any inconsistencies between the values, so I think using lastName should be fine. Just would be something to double check in the future but I think it works for this exercise.

Reply:
I realized this after submitting hahaha. I even checked and said to myself that the column exists, use it... and I didn't. Good to know that it didn't change the outcome too too much